pkg/pillar/cmd/zedmanager/handleclusterapp.go

```diff
@@ -40,12 +40,14 @@ func handleENClusterAppStatusImpl(ctx *zedmanagerContext, key string, status *ty
 				log.Errorf("handleENClusterAppStatusImpl(%s) AppInstanceConfig missing for app", key)
 				return
 			}
-			// XXX this will be handled in later PR in clustering and zedmanager code
-			//handleCreateAppInstanceStatus(ctx, *aiConfig)
+			handleCreateAppInstanceStatus(ctx, *aiConfig)
 		} else {
-			// re-publish the aiStatus, in case the cluster status has changed.
-			log.Functionf("handleENClusterAppStatusImpl(%s) for app-status %v aiStatus %v", key, status, aiStatus)
+
+			activateAIStatusUUID(ctx, key)
+			// We need to modify and publish the aiStatus with NoUploadStatsToController field set.
 			publishAppInstanceStatus(ctx, aiStatus)
+			publishAppInstanceSummary(ctx)
```

Comment:
You made activateAIStatusUUID() publish the status and summary, so no need to do it here.
In fact it would be incorrect to publish the Status when that function removes the AIStatus.

Reply:
oh on the second thought I should make  activateAIStatusUUID() to publish status and summary irrespective of changed or not because we need to publish to controller the app metrics when app gets scheduled on this node.


So I will remove publish here, and unconditionally publish in activateAIStatusUUID()